rfc/egress-traffic.md

```diff
@@ -0,0 +1,238 @@
+
+- **No Subscription Requirement:** Billing Meters allow usage tracking for customers without needing an active subscription, simplifying pre-subscription billing.
+- **Real-Time Reporting:** Automatically tracks and aggregates usage events, ensuring accurate and timely reporting.
+- **Scalability:** Optimized for high-volume usage reporting, making it ideal for Storacha Network’s needs.
+- **No Need for Immediate Migration:** Start tracking egress traffic with the Billing Meter and create new usage-based prices later.
+- **Adding New Prices for Each Product:** Create new prices for each of the three products (**Starter, Lite, and Business**), based on tiers of egress data usage. Once associated with the Billing Meter, Stripe will automatically track usage for billing.
+
+### Example Pricing Tiers:
+
+1. **Egress Starter:**
+
+   - **0 to 5 GB:** Free
+   - **5 GB to 20 GB:** $0.10 per GB
+   - **20 GB+:** $0.07 per GB
+
+2. **Egress Lite:**
+
+   - **0 to 10 GB:** Free
+   - **10 GB to 50 GB:** $0.08 per GB
+   - **50 GB+:** $0.06 per GB
+
+3. **Egress Business:**
+   - **0 to 20 GB:** Free
+   - **20 GB to 100 GB:** $0.05 per GB
+   - **100 GB+:** $0.03 per GB
+
+### Possible Solution
+
+Since **Usage Records** are being deprecated by Stripe and **Billing Meters** offer greater flexibility and scalability, adopting **Billing Meters** for Storacha Network’s usage-based billing is recommended.
+
+### Benefits of Billing Meters:
+
+- **No Customer Disruption:** Existing billing remains intact while usage is tracked in the background through the Billing Meter.
+- **Flexible Tiers:** New prices can be created for each product tier and associated with the Billing Meter for automated usage tracking.
+- **Scalability:** This approach ensures that billing for egress traffic is accurate and scalable with minimal operational overhead.
+
+## Questions & Answers
+
+### 1. **Operational Complexity:**
+
+- **Question:** Which approach requires less effort to maintain in the long run? Can the additional infrastructure of Alternative 1 or 3 be justified given their decoupled architecture?
+- **Answer:** Alternative 2 requires less effort to maintain in the long run since it leverages existing infrastructure (`w3up/upload-api`), avoiding the need to set up new infrastructure components (SQS, Lambda, DynamoDB).
+
+### 2. **Security:**
+
+- **Question:** How should delegation and authentication mechanisms be handled, particularly in Alternative 2? What security implications arise from allowing the Cloudflare Freeway Worker to directly interact with the `w3up/upload-api`?
+- **Answer:** Delegation can be handled by granting the Cloudflare Freeway Worker permission to invoke the required capabilities. The Gateway will need to become a `w3up` client to claim the delegation and execute the invocation securely.
+- **Question:** Who will create the delegation to the Cloudflare Freeway Worker?
+- **Answer:** ???
```

Comment:
we can do this manually using the upload api service's private key

Reply:
That makes sense @travis .